src/backend/requirements.in

```diff
@@ -18,6 +18,7 @@ django-maintenance-mode                 # Shut down application while reloading
 django-markdownify                      # Markdown rendering
 django-mptt                             # Modified Preorder Tree Traversal
 django-markdownify                      # Markdown rendering
+django-middleware-global-request        # Global request middleware
```

Comment:
Are you sure about this? This bleeds possibly sensitive request info into the python thread. Not sure how to shield this from plugins

Reply:
I'm not sure about it - it feels like a pretty big hammer for the problem.

However caching this info against the request saves a heap of DB hits..

Actually thinking about it some more, the *better* way is to get a global cache working. The current plugin registry hash is coded around the fact that the default Django cache does not share between processes. If we had a global cache (redis or memcached) we could bypass this entirely.

The global request approach showed that there is a potential speed boost to be had, but I agree it's not the best approach.